base/broadcast.jl

```diff
@@ -440,4 +440,24 @@ for (sigA, sigB) in ((BitArray, BitArray),
     end
 end
 
+############################################################
+
+# x[...] .= f.(y...) ---> broadcast!(f, dotview(x, ...), y...).
+# The dotview function defaults to getindex, but we override it in
+# a few cases to get the expected in-place behavior without affecting
+# explicit calls to view.   (All of this can go away if slices
+# are changed to generate views by default.)
+
+dotview(args...) = getindex(args...)
+dotview(A::AbstractArray, args...) = view(A, args...)
+dotview{T<:AbstractArray}(A::AbstractArray{T}, args...) = getindex(A, args...)
```

Comment:
You're right, it should be `view` also.  I was thinking that `view` wasn't necessary, since `A[i,j]` returns a mutable object.